### PR TITLE
Only update tagged_at for status changes when post is completed

### PR DESCRIPTION
**Pull request by teceler**

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -310,7 +312,7 @@ def set_timestamps
     return if skip_edited
     self.edited_at = self.updated_at
     return if skip_tagged
-    return if replies.exists? && !status_changed?
+    return if replies.exists? && status != Post::STATUS_COMPLETE
```
Comment (Marri) [inline app/models/post.rb]: I think you might still need to check if the status changed here, as well as what it is?

File: app/models/post.rb [hunk 2]
```diff
@@ -200,7 +200,9 @@ def read_time_for(viewing_replies)
 
     # testing for case where the post was changed in status more recently than the last reply
     audits_since_last_reply = audits.where('created_at > ?', most_recent.created_at)
-    audit = audits_since_last_reply.detect { |a| a.audited_changes.key?('status') }
+    audit = audits_since_last_reply.detect do |aud|
```
Comment (Marri) [inline app/models/post.rb]: plausibly we can do something fancier with the actually queryable column here

File: app/models/post.rb (hunk 1 above)
Comment (teceler) [reply app/models/post.rb]: ...oh. yes. I did... not entirely think that through, yeah xD

File: app/models/post.rb [hunk 3]
```diff
@@ -199,9 +199,9 @@ def read_time_for(viewing_replies)
     return most_recent.updated_at if most_recent.updated_at > edited_at
 
     # testing for case where the post was changed in status more recently than the last reply
-    audits_since_last_reply = audits.where('created_at > ?', most_recent.created_at)
-    audit = audits_since_last_reply.detect { |a| a.audited_changes.key?('status') }
-    return most_recent.updated_at unless audit
+    audits_exist = audits.where('created_at > ?', most_recent.created_at).where(action: 'update')
+    audits_exist = audits_exist.where("(audited_changes -> 'status' ->> 1)::integer = ?", Post::STATUS_COMPLETE).exists?
+    return most_recent.updated_at unless audits_exist
```
Comment (Marri) [inline app/models/post.rb]: for viewport reasons I propose:

    audits = audits.where('created_at > ?', most_recent.created_at).where(action: 'update')
    audits = audits_exist.where("(audited_changes -> 'status' ->> 1)::integer = ?", Post::STATUS_COMPLETE).exists?
    return most_recent.updated_at unless audits.exists?

Comment (teceler) [reply app/models/post.rb]: ...you've got .exists? there twice, but

Comment (Marri) [reply app/models/post.rb]: YOU SAW NOTHING NOTHING HAPPENED

**Pull request merged**